server/src/main/java/org/elasticsearch/search/searchafter/SearchAfterBuilder.java

```diff
@@ -184,7 +184,8 @@ private static Object convertValueFromSortType(String fieldName, SortField.Type
                     if (value instanceof Number) {
                         return ((Number) value).longValue();
                     }
-                    return Long.parseLong(value.toString());
+                    return format.parseLong(value.toString(), false,
+                        () -> { throw new IllegalStateException("now() is not allowed in [search_after] key"); });
```

Comment:
This change seems not to be related with the changes in this PR?

Reply:
It is related since we now use search after to build the index sorted query but you're right that this is also a bug fix so I can extract this change in a specific pr if you prefer.

Reply:
I don't think is necessary. My only concern is that this brings a change that later on is difficult to track. For me it is ok if you want to leave it here.

Reply:
+1 to leave it here, the change is simple and has dedicated unit tests.